### DISCUSSION TITLE
Drone Toilet Bowling

### DISCUSSION THREAD

**Discussion by Some-Won-1** [iNavFlight/inav#8863] [Ideas]
Assuming the GPS is working correctly, should not a Sanity Check be performed against the Mag? If there is a problem then Fail-safe should activate. There is no reason for allowing a toilet bowel condition to continue! This kind of fly away tells me something is not ready for prime time.

**Comment by stronnag** [comment]
It is basic setup. It's been well documented on the wiki for years.

**Comment by DzikuVx** [comment]
Toilet bowling is always cause by wrong magnetometer alignment of placement

**Comment by breadoven** [comment]
I did do something a while back to sanity check the mag heading by comparing it with GPS ground course. It did sort of work but obviously has problems when the quad isn't flying forward even when you take account of this by correcting for control inputs. I did think about checking for toilet bowling which is easier but decided you can do that with your eyes by checking position hold immediately after takeoff. I suppose having a sanity check for this would help if the mag loses its calibration in flight but how likely is that to happen ? Total Mag loss will result in an emergency landing in most cases due to loss of valid heading except seemingly during PosHold since it only appears to check for a valid heading on initiating PosHold but not after. Perhaps this could be improved.
